plugins/BEdita/Core/src/Model/Action/UpdateRelatedObjectsAction.php

```diff
@@ -74,23 +42,27 @@ public function execute(array $data = [])
      * using `Tree` entity with `ParentObjects` association assuring to
      * always use a "to one" relation.
      *
-     * @param \Cake\Datasource\EntityInterface $entity The starting entity.
-     * @return \Cake\Datasource\EntityInterface
+     * @param array $data Action data.
+     * @return array
      */
-    protected function getEntity(EntityInterface $entity)
+    protected function prepareData(array $data)
     {
-        if (!($entity instanceof Folder) || $this->Association->getName() !== 'Parents') {
-            return $entity;
+        if (empty($data['entity']) || !($data['entity'] instanceof Folder) || $this->Association->getName() !== 'Parents') {
+            return $data;
         }
 
         $table = $this->Association->junction();
         $entity = $table->find()
-            ->where([$table->association('Objects')->getForeignKey() => $entity->id])
+            ->where([$table->association('Objects')->getForeignKey() => $data['entity']->id])
```

Comment:
Do we need an additional check on `$data['entity']` to verify that it's an `Entity` or it is unnecessary?

Reply:
If `!($data['entity'] instanceof Folder)` this method returns `$data` as is (see rows 50-52).

Reply:
ok, I missed it ;)